File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    
    res = []
    for i in s:
        if i == "(":
            res.append(i)
        else:
            if res:
                res.pop()
            else:
                answer = False
                break
    if res:
        answer = False
    return answer

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        boolean answer = false;

        Stack<Character> stack = new Stack<>();
        for(int i=0; i<s.length(); i++) {
            stack.add(s.charAt(i));
        }
        
        char start = stack.pop();
        if(start == '(') {
            return answer;
        }
        
        int count = 1;
        while(!stack.isEmpty()) {
            char c = stack.pop();
            switch(c) {
                case ')':
                    count++;
                    break;
                case '(':
                    count--;
                    break;
            }

            if(count < 0) {
                return answer;
            }
        }
        
        if(count == 0) {
            answer = true;
        }
        return answer;
    }
}